### PR TITLE
ci: Add a nightly CI job for integration tests

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: .github/workflows/nightly-ci.yml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        if: ${{ always() }}  # this will guarantee that step won't be canceled and resources won't leak
+        env:
+          FEAST_USAGE: "False"
+          IS_TEST: "True"
+          SNOWFLAKE_CI_DEPLOYMENT: ${{ secrets.SNOWFLAKE_CI_DEPLOYMENT }}
+          SNOWFLAKE_CI_USER: ${{ secrets.SNOWFLAKE_CI_USER }}
+          SNOWFLAKE_CI_PASSWORD: ${{ secrets.SNOWFLAKE_CI_PASSWORD }}
+          SNOWFLAKE_CI_ROLE: ${{ secrets.SNOWFLAKE_CI_ROLE }}
+          SNOWFLAKE_CI_WAREHOUSE: ${{ secrets.SNOWFLAKE_CI_WAREHOUSE }}
+        run: pytest -n 8 --cov=./ --cov-report=xml --verbose --color=yes sdk/python/tests --integration --durations=5
```
Comment (kevjumba) [inline .github/workflows/nightly-ci.yml:114]: Are we going to add the ci contrib tests in a separate pr?

Comment (achals) [reply .github/workflows/nightly-ci.yml:114]: yeah I want to do it in a separate PR.

File: .github/workflows/nightly-ci.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      - uses: actions/checkout@v2
+        with:
+          ref: develop
+      - id: check_date
+        name: Check if there were commits in the last day
+        if: ${{ github.event_name == 'schedule' }}
+        run: echo '::set-output name=WAS_EDITED::'$(test -n "$(git log --format=%H --since='24 hours ago')" && echo 'true' || echo 'false')
+
+  integration-test-python:
+    # all jobs MUST have this if check for 'ok-to-test' or 'approved' or 'lgtm' for security purposes.
```
Comment (adchia) [inline .github/workflows/nightly-ci.yml:28]: unnecessary comment?

**Review by adchia — APPROVED** [review]

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**